src/fastalloc/vregset.rs

```diff
@@ -1,104 +1,86 @@
 use core::fmt;
 
-use crate::{RegClass, VReg};
+use crate::ion::data_structures::VRegIndex;
+use crate::VReg;
 use alloc::vec;
 use alloc::vec::Vec;
 
 #[derive(Clone)]
 struct VRegNode {
-    next: u32,
-    prev: u32,
-    class: RegClass,
+    next: VRegIndex,
+    prev: VRegIndex,
+    vreg: VReg,
 }
 
 // Using a non-circular doubly linked list here for fast insertion,
```

Comment:
This comment needs to be updated.

Reply:
Yeah,  I'm still getting to all these things. I'm not yet done making changes